xtask/codegen/markdown.ungram

```diff
@@ -58,6 +58,13 @@ AnyContainerBlock = MdQuote | MdBulletListItem | MdOrderListItem
 // h1..h6
 MdHeader = before:MdHashList MdParagraph? after:MdHashList
 
+MdHeader1 = before:MdHashList MdParagraph? after:MdHashList
+MdHeader2 = before:MdHashList MdParagraph? after:MdHashList
+MdHeader3 = before:MdHashList MdParagraph? after:MdHashList
+MdHeader4 = before:MdHashList MdParagraph? after:MdHashList
+MdHeader5 = before:MdHashList MdParagraph? after:MdHashList
+MdHeader6 = before:MdHashList MdParagraph? after:MdHashList
```

Comment:
This change doesn't quite look right to me. We already have `MdHeader` defined above, but you've added 6 new nodes for headers. IMO, we should rename `MdHeader` into `AnyMdHeader`, and have it be a union of all the other header levels.

Reply:
okay, but how should we represent the level of the heading ?

Reply:
Having `MdHeader1` through `MdHeader6` is fine. I'm saying we should change `MdHeader` to:

```
AnyMdHeader = MdHeader1 | MdHeader2 | MdHeader3 | MdHeader4 | MdHeader5 | MdHeader6
```